metric_source/local/local.go

```diff
@@ -98,15 +78,40 @@ func (local *Local) Fetch(target string, from int64, until int64, allowRealTimeA
 				})
 			}
 			result.Metrics = append(result.Metrics, metrics...)
-			for _, pattern := range patterns {
-				result.Patterns = append(result.Patterns, pattern.Metric)
+			for _, mr := range metricRequests {
+				result.Patterns = append(result.Patterns, mr.Metric)
 			}
 		}
 	}
 
 	return result, nil
 }
 
+func evalExpr(target string, expression parser.Expr, from, until int64, metricsMap map[parser.MetricRequest][]*types.MetricData) (result []*types.MetricData, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			result = nil
+			err = ErrEvaluateTargetFailedWithPanic{target: target, recoverMessage: r, stackRecord: debug.Stack()}
+		}
+	}()
+	result, err = expr.EvalExpr(context.Background(), expression, from, until, metricsMap)
+	if err != nil {
+		if errors.Is(err, parser.ErrSeriesDoesNotExist) {
+			// It is ok for us because some clients of Moira send events instead of metrics
```

Comment:
let's remove it

Reply:
do you mean remove comment? You are highlighted 97-100 lines

Reply:
only comment

Reply:
why? :)

Reply:
moved description to commit message and PR body